app/models/cloud_volume_snapshot.rb

```diff
@@ -26,6 +26,61 @@ def my_zone
     self.class.my_zone(ext_management_system)
   end
 
+  def self.create_snapshot_queue(userid, cloud_volume, options = {})
+    ext_management_system = cloud_volume.&ext_management_system
+    task_opts = {
+      :action => "creating volume snapshot in #{ext_management_system.inspect} for #{cloud_volume.inspect} with #{options.inspect}",
+      :userid => userid
+    }
+
+    queue_opts = {
+      :class_name  => cloud_volume.class.name,
+      :instance_id => cloud_volume.id,
+      :method_name => 'create_volume_snapshot',
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => 'ems_operations',
+      :zone        => my_zone(ext_management_system),
+      :args        => [options]
+    }
+
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def self.create_snapshot(cloud_volume, options)
+    raw_create_snapshot(cloud_volume, options)
+  end
+
+  def self.raw_create_snapshot(cloud_volume, options)
+    raise NotImplementedError, _("create_snapshot must be implemented in a subclass")
```

Comment:
Super minor, but you know someone is going to read the exception and implement `create_snapshot` instead of `raw_create_snapshot` in their subclass